test/test_base.py

```diff
@@ -231,6 +234,36 @@ def test_type_cast(self, linear_type: LinearType, linear_dtype: torch.dtype):
             y.dtype == torch.bfloat16
         ), f"y.dtype is {y.dtype}, expected {torch.bfloat16}"
 
+    @pytest.mark.parametrize("use_compile", [False, True])
+    def test_weight_caching(self, use_compile):
+        M, K, N = 16, 32, 64
+        dtype = torch.bfloat16
+        config.allocate_float8_weight_cache_buffers = True
+
+        x = torch.randn(M, K, device="cuda", dtype=dtype)
+        m_ref = nn.Linear(K, N, bias=True, device="cuda", dtype=dtype)
+        m = Float8Linear.from_float(copy.deepcopy(m_ref), emulate=False)
+
+        if use_compile:
+            m = torch.compile(m)
+
+        config.weight_cache_enabled = False
+
+        y1 = m(x)
+        y1.sum().backward()
+        grad1 = m.weight.grad.clone().detach()
+
+        config.weight_cache_enabled = True
```

Comment:
I was also thinking about this after you brought this up: https://github.com/pytorch-labs/float8_experimental/commit/c09b67ffee21ffd1ac361ce9c0114e8eadd852bb

was more curious as to what the ideal ux would be for turning on and off the weight-cachihng

Reply:
I think right now it has to be "what's the UX which isn't terribly slow after torch.compile".  I tried a couple of things and the global variable was the only one which didn't have a large overhead :(

TE requires user to explicitly pass a `is_first_microbatch` flag, I didn't do that because that requires a model rewrite.  The global thing allows the signature of `Float8Linear`'s forward to stay the same.